### PR TITLE
Local activity nonfirst attempt counts for metering

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Count non-first-WFT attempts during each WFT for each local activity

## Why?
In support of recent metering changes

## Checklist

1. Closes 

2. How was this tested:
Added integ test

3. Any docs updates needed?
Nothing Go specific

### CODE REVIEW

File: internal/internal_event_handlers.go [hunk 1]
```diff
@@ -871,6 +874,32 @@ func (wc *workflowEnvironmentImpl) GetRegistry() *registry {
 	return wc.registry
 }
 
+// ResetLAWFTAttemptCounts resets the number of attempts in this WFT for all LAs to 0 - should be
+// called at the beginning of every WFT
+func (wc *workflowEnvironmentImpl) ResetLAWFTAttemptCounts() {
```
Comment (Quinn-With-Two-Ns) [inline internal/internal_event_handlers.go:879]: Can we write an integration test to cover this function as well? I think `TestSendsCorrectMeteringData` would still pass even if this function wasn't called.

File: test/integration_test.go [hunk 2]
```diff
@@ -3693,6 +3693,53 @@ func (ts *IntegrationTestSuite) TestScheduleUpdateWorkflowActionMemo() {
...
+	ts.Nil(w.Start())
+
+	wfOpts := ts.startWorkflowOptions("test-sends-correct-metering-data")
+	wfOpts.WorkflowTaskTimeout = 2 * time.Second
+	ts.NoError(ts.executeWorkflowWithOption(wfOpts,
+		ts.workflows.WorkflowWithLocalActivityRetriesAndHeartbeat, nil))
+
+	ts.Equal(uint32(0), nonfirstLAAttemptCounts[0])
+	for i := 1; i < len(nonfirstLAAttemptCounts); i++ {
+		ts.True(nonfirstLAAttemptCounts[i] > 0)
```
Comment (Quinn-With-Two-Ns) [inline test/integration_test.go:3738]: Can we not test the exact count?

Comment (Sushisource) [reply test/integration_test.go:3738]: We could, but there's a good chance it'd make it flaky I think. How many attempts happen per WFT is going to be somewhat timing sensitive.

File: internal/internal_event_handlers.go (hunk 1 above)
Comment (Sushisource) [reply internal/internal_event_handlers.go:879]: It fails if the call is removed - otherwise the attempts which span a heartbeat are over-counted.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Review by cretz — APPROVED** [review]

**Pull request merged**